source/lac/scalapack.cc

```diff
@@ -0,0 +1,966 @@
+{
+  if (mpi_process_is_active)
+    Cblacs_gridexit(blacs_context);
+
+  MPI_Comm_free(&mpi_communicator_inactive_with_root);
+}
+
+
+
+unsigned int ProcessGrid::get_process_grid_rows() const
+{
+  return n_process_rows;
+}
+
+
+
+unsigned int ProcessGrid::get_process_grid_columns() const
+{
+  return n_process_columns;
+}
+
+
+
+bool ProcessGrid::is_process_active() const
+{
+  return mpi_process_is_active;
+}
+
+
+
+template <typename NumberType>
+void ProcessGrid::send_to_inactive(NumberType *value, const int count) const
+{
+  Assert (count>0, ExcInternalError());
+  if (mpi_communicator_inactive_with_root != MPI_COMM_NULL)
+    {
+      const int ierr =
+        MPI_Bcast(value,count,
+                  Utilities::MPI::internal::mpi_type_id (value),
+                  0/*from root*/,
+                  mpi_communicator_inactive_with_root);
+      AssertThrowMPI(ierr);
+    }
+}
+
+
+
+template <typename NumberType>
+ScaLAPACKMatrix<NumberType>::ScaLAPACKMatrix(const size_type n_rows_,
+                                             const size_type n_columns_,
+                                             const std::shared_ptr<const ProcessGrid> process_grid,
+                                             const size_type row_block_size_,
+                                             const size_type column_block_size_,
+                                             const LAPACKSupport::Property property)
+  :
+  TransposeTable<NumberType> (),
+  state (LAPACKSupport::unusable),
+  property(property),
+  grid (process_grid),
+  n_rows(n_rows_),
+  n_columns(n_columns_),
+  row_block_size(row_block_size_),
+  column_block_size(column_block_size_),
+  uplo('L'), // for non-symmetric matrices this is not needed
+  first_process_row(0),
+  first_process_column(0),
+  submatrix_row(1),
+  submatrix_column(1)
+{
+  Assert (row_block_size > 0,
+          ExcMessage("Row block size has to be positive."));
+  Assert (column_block_size > 0,
+          ExcMessage("Column block size has to be positive."));
+  Assert (row_block_size <= n_rows,
+          ExcMessage("Row block size can not be greater than the number of rows of the matrix"));
+  Assert (column_block_size <= n_columns,
+          ExcMessage("Column block size can not be greater than the number of columns of the matrix"));
+
+  if (grid->mpi_process_is_active)
+    {
+      // Get local sizes:
+      n_local_rows = numroc_(&n_rows, &row_block_size, &(grid->this_process_row), &first_process_row, &(grid->n_process_rows));
+      n_local_columns = numroc_(&n_columns, &column_block_size, &(grid->this_process_column), &first_process_column, &(grid->n_process_columns));
+
+      // LLD_A = MAX(1,NUMROC(M_A, MB_A, MYROW, RSRC_A, NPROW)), different between processes
+      int lda = std::max(1,n_local_rows);
+
+      int info=0;
+      descinit_(descriptor, &n_rows, &n_columns, &row_block_size, &column_block_size,&first_process_row,&first_process_column,&(grid->blacs_context), &lda, &info);
```

Comment:
Can you wrap this line. It will be easier to look at the code on Github later on :-)